content/en/docs/tasks/access-kubernetes-api/custom-resource-definition-versioning.md

```diff
@@ -0,0 +1,232 @@
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  # name must match the spec fields below, and be in the form: <plural>.<group>
+  name: crontabs.example.com
+spec:
+  # group name to use for REST API: /apis/<group>/<version>
+  group: example.com
+  # list of versions supported by this CustomResourceDefinition
+  versions:
+  - Name: v1beta1
+    # Each version can be enabled/disabled by Served flag.
+    Served: true
+    # One and only one version must be marked as the storage version.
+    storage: true
+  - Name: v1
+    Served: true
+    storage: false
+  # either Namespaced or Cluster
+  scope: Namespaced
+  names:
+    # plural name to be used in the URL: /apis/<group>/<version>/<plural>
+    plural: crontabs
+    # singular name to be used as an alias on the CLI and for display
+    singular: crontab
+    # kind is normally the CamelCased singular type. Your resource manifests use this.
+    kind: CronTab
+    # shortNames allow shorter string to match your resource on the CLI
+    shortNames:
+    - ct
+```
+
+You can save the CustomResourceDefinition in a YAML file, then use
+`kubectl create` to create it.
+
+```shell
+kubectl create -f my-versioned-crontab.yaml
+```
+
+After creation, the API server starts to serve each enabled version at its
+resolved endpoint.. In the
+above example, the APIs are available at `/apis/example.com/v1beta1` and
+`/apis/example.com/v1`.
+
+### Version order
+
+The _order_ of the versions in the CustomResourceDefinition defines a priority
+for each version. The version that comes first in term of priority is used by
+kubectl as the default version to access objects. 
+
+The version _name_ in the CustomResourceDefiniton version list is used to
+compute the order of versions, using the following algorithm to attempt to parse
+the versions by version, then stability level.
+
+- Versions starting with a `v` are sorted above those that are not. Other string
+  prefixes are sorted alphanumerically.
+- Within `v` versions:
+  - If a number comes next, the numbers are sorted from largest to smallest.
+  - If the strings `GA`, `beta`, `alpha` follow the number, they are sorted in
+    that order.
+      - If another number follows the `GA`, `beta`, or `alpha`, those numbers
+        are sorted from largest to smallest.
+
+This might make sense if you look at the following sorted version list:
+
+```none
+- v10
+- v2
+- v1
+- v11beta2
+- v10beta3
+- v3beta1
+- v12alpha1
+- v11alpha2
+- foo1
+- foo10.
+```
+
+For the example in [Specify multiple versions](#specify-multiple-versions), the
+version sort order is `v1`, followed by `v1beta`. This causes the kubectl
+command to use `v1` as the default version unless the provided object specifies
+the version.
+
+## Manage object versions
+
+The only conversion the API server performs on a CustomResourceDefinition is to
+fix the apiVersion of the object. When an object is requested, the version
```

Comment:
confusing: on request an object is not persisted in general (e.g. on GET, LIST, WATCH).

Better: An object is stored in etcd with the storage version at the time of creation. When the object is requested, the API server uses returns it in the API version of the request. In the case of CREATE or UPDATE requests it will persist the object with the current storage version, not by the request version.

Reply:
Thanks, I didn't like how this turned out either. I was doing my best with the starting text. This is really good feedback.